addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -103,6 +103,22 @@ registerModel({
             const { length, [length - 1]: lastMessageView } = this.messageViews;
             return lastMessageView ? replace(lastMessageView) : clear();
         },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageListView() {
+            if (!this.thread || !this.threadCache) {
+                return clear();
+            }
+            if (this.isLoading && !this.threadCache.isLoaded) {
+                return clear();
+            }
+            if (this.threadCache.isLoaded || this.thread.isTemporary) {
+                return insertAndReplace();
+            }
+            return clear();
```

Comment:
But at least clean the conditions here, there should be a single boolean expression that determines with `? :` whether it's `insertAndReplace` or `clear`

I think the middle condition is actually useless? Because `this.threadCache.isLoaded` appears in all conditions, and `isLoading` will never be set if `this.thread.isTemporary`.

So in summary we have a message list if the thread cache is loaded, or if it's a temporary thread (loading concepts do not apply), which is precisely the last condition. Whether the cache is loading or not for temporary thread is irrelevant, just as much as it is irrelevant if the thread cache has already been loaded once.